src/common/meta/src/ddl/alter_table.rs

```diff
@@ -182,23 +190,33 @@ impl AlterTableProcedure {
 
     pub async fn submit_alter_region_requests(&mut self) -> Result<Status> {
         let table_id = self.data.table_id();
+        let table_route_manager = self.context.table_metadata_manager.table_route_manager();
 
-        let table_route = self
-            .context
-            .table_metadata_manager
-            .table_route_manager()
+        let table_route = table_route_manager
             .get(table_id)
             .await?
             .context(TableRouteNotFoundSnafu { table_id })?
             .into_inner();
-        let region_routes = table_route.region_routes()?;
+        let region_routes = match table_route {
+            TableRouteValue::Physical(x) => x.region_routes,
+            TableRouteValue::Logical(x) => {
+                let physical_table_id = x.physical_table_id();
+                let physical_table_route = table_route_manager
+                    .get(physical_table_id)
+                    .await?
+                    .context(TableRouteNotFoundSnafu {
+                        table_id: physical_table_id,
+                    })?;
+                physical_table_route.region_routes()?.clone()
+            }
+        };
```

Comment:
It's okay for now. Let's move these methods into `table_meatdata_manager' later.

Reply:
Let  me do it now